src/plugins/opensearch_dashboards_react/public/table_list_view/components/create_button_menu.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import React, { useState } from 'react';
+import {
+  EuiButton,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiFlexItem,
+  EuiPopover,
+} from '@elastic/eui';
+import type { DashboardCreator, DashboardCreatorFn, DashboardCreators } from '../../context/types';
+
+interface CreateButtonProps {
+  // createItem: () => {};
+  dashboardItemCreatorClickHandler: (creatorFn: DashboardCreatorFn) => (event: any) => void;
+  dashboardItemCreators: () => DashboardCreators;
+}
+
+const CreateButton = (props: CreateButtonProps) => {
+  const [isPopoverOpen, setPopover] = useState(false);
+
+  const onButtonClick = () => {
+    setPopover(!isPopoverOpen);
+  };
+
+  const closePopover = () => {
+    setPopover(false);
+  };
+
+  const getItems = (creators: DashboardCreators) => {
+    return creators.map((creator: DashboardCreator) => (
+      <EuiContextMenuItem
+        key={creator.id}
+        onClick={props.dashboardItemCreatorClickHandler(creator.creatorFn)}
+      >
+        {creator.defaultText}
+      </EuiContextMenuItem>
+    ));
+  };
+
+  const forceSingleButtonOnly = false;
+
+  const creators: DashboardCreators = props.dashboardItemCreators();
+
+  const renderMenu = () => {
+    if (creators.length === 0) {
+      return null;
+    } else if (forceSingleButtonOnly || creators.length === 1) {
+      const creator = creators[0];
+      return (
+        <EuiFlexItem grow={false}>
+          <EuiButton
+            onClick={props.dashboardItemCreatorClickHandler(creator.creatorFn)}
+            data-test-subj="newItemButton"
+            iconType="plusInCircle"
+            fill
+          >
+            Create {creator.defaultText}
+          </EuiButton>
+        </EuiFlexItem>
+      );
+    } else {
+      const button = (
+        <EuiButton iconType="arrowDown" iconSide="right" onClick={onButtonClick} fill>
+          Create
```

Comment:
```suggestion
        <FormattedMessage
              id="opensearch-dashboards-react.tableListView.listing.createNewItemButtonLabel"
              defaultMessage="Create {entityName}"
              values={{ entityName: this.props.entityName }}
            />
```

Reply:
@ashwin-pc The suggested <FormattedMessage> assumes the given item to specifically be exactly the Dashboards-Core create item.   This not guaranteed in code, so placing this formatter here does not agree with the arbitrary item.